workflow-testing/src/test/java/com/squareup/workflow1/testing/WorkerTesterTest.kt

```diff
@@ -15,13 +15,28 @@ import kotlin.test.assertFalse
 @OptIn(ExperimentalCoroutinesApi::class)
 class WorkerTesterTest {
 
-  @Test fun `assertNoOutput passes after worker finishes without emitting`() {
-    val worker = Worker.finished<Unit>()
+  @Test fun `assertNoOutput succeeds in live flow without output`() {
+    val worker = Worker.create<Unit> {
+      suspendCancellableCoroutine { }
+    }
     worker.test {
       assertNoOutput()
     }
   }
 
+  @Test fun `assertNoOutput fails after worker finishes without emitting`() {
```

Comment:
if worker finishes without emitting anything, then `assertNoOutput` should succeed right? as the name suggest it should check for no output

Reply:
@steve-the-edwards I'm confused by this too.

Reply:
I'm confused by two things: I think @RotBolt is right that the method name is wrong; and I'm unclear why we want this semantic.

Why is it an error for a worker to complete when we `assertNoOutput()`? This change broke some existing tests that were building workers via `Observable.just()`, for example.

Reply:
I think you guys are right I should have made more noise about the change in this semantic. I just took the easy way out and inherited the change from `expectNoEvents()` when it started throwing on Completion after Turbine 0.3.0.

I've done a quick look and I'm not sure there is good way to reproduce the behavior we had before. @RotBolt could you create an issue to update this or rename the assertion to something like `assertNoOutputOrCompletion()`?

I think I am ok with having places where we use flowOf { 1 } or Observable.just(false) to start using `assertFinished()` instead of `assertNoOutput()`. But maybe that's a cop out?

Reply:
It's easy to replace `Observable.just(false)` with `BehaviorRelay.createDefault(false)`. Is there a similar idiom for `Flow`?

Reply:
```
flow { emit(false); suspendCancellableCoroutine {} }
```

so tidy right? 

hmm. must be something better